my-cart/src/app/reducers/products.reducer.ts

```diff
@@ -0,0 +1,47 @@
+import { Product } from "../model/Products.model";
+import { ProductState } from "../products/ProductState";
+import * as productActionTypes from './../products/product-actions-types';
+
+
+const InitialState:ProductState = {
+   categories : [],
+   addedProducts : null
+}
+
+export function productReducer(state=InitialState,action:{type:string,payload?:any}){
+      let newCartItems: any;
+      let increasedQuantity;
+      switch(action.type){
+          case productActionTypes.ADD_CATEGORIES :
+              return {...state,categories:action.payload}
+
+          case productActionTypes.ADD_CART_ITEMS :
+              newCartItems = { ...state.addedProducts,[action.payload.id] : action.payload};
+              return {...state,addedProducts:newCartItems}
+
+          case productActionTypes.ADD_CART_QUANTITY : 
```

Comment:
May be something like `INC_CART_ITEM_QTY`, `DEC_CART_ITEM_QTY` will make more semantical sense (increment, decrement)

Reply:
Done and pushed